### PR TITLE
feat: Support Spark 3.5.1

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

For https://github.com/orgs/stackabletech/discussions/42

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

**Comment by sbernauer** [comment]
Started https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/spark-k8s-operator-it-custom/128/

File: docs/modules/spark-k8s/partials/supported-versions.adoc [hunk 1]
```diff
@@ -3,6 +3,7 @@
 // Stackable Platform documentation.
 // Please sort the versions in descending order (newest first)
 
-- 3.5.0 (Hadoop 3.3.4, Scala 2.12, Python 3.11, Java 11)
+- 3.5.1 (Hadoop 3.3.4, Scala 2.12, Python 3.11, Java 11)
```
Comment (NickLarsenNZ) [inline docs/modules/spark-k8s/partials/supported-versions.adoc:6]: I guess we want to bump Hadoop to 3.3.6 here?

**Review by NickLarsenNZ — APPROVED** [review]
LGTM. Comment about Hadoop 3.3.6

File: docs/modules/spark-k8s/partials/supported-versions.adoc (hunk 1 above)
Comment (sbernauer) [reply docs/modules/spark-k8s/partials/supported-versions.adoc:6]: Good point, I was thinking the same :+1: 
TLDR: @maltesander decided in https://github.com/stackabletech/docker-images/commit/b407c0cbf4f4e1f98b3c009363928144e25a1d8e to not bump it, I'm happy to stick to that and bump it in the next release.

Also our somewhat recommended product version (we don't have that yet officially) is probably 3.3.4, as 3.3.6 [has problems](https://github.com/stackabletech/hdfs-operator/issues/440), so I guess this would be the matching HDFS client version. But all of this are no strong opinions, I just have the feeling the larger portion of our developers would prefer to not change it just days before the release.

But we can bring this up during daily

Comment (maltesander) [reply docs/modules/spark-k8s/partials/supported-versions.adoc:6]: Yeah no strong opinion from my side, just that we did not figure out (yet?) what is not working with 3.3.6. for some.
